### DISCUSSION TITLE
How to embebe Birt report in vanila html/css/javascript page?

### DISCUSSION THREAD

**Discussion by wagnermarques** [eclipse-birt/birt#1763] [Q&A]
Hi folks.
 
Can someone help, please, to figure out some alternative to iframe to show a specific birt report in my vanila html/css/javascript page?

 I've been searched  asked to AIs, but the response was about birt.viewer.js and  jsapi but it's no worked. 
 
 I'm beginning with Birt report solution.  Can someone help shed some light to acomplish this taks, please? 

Best regards.

**Comment by SteveSchafer-Innovent** [comment]
You can use XMLHttpRequest to send a GET request and then set the response string  to the innerHTML of some element in your page.  If your web page exists in a different domain from your BIRT report provider, you could get a CORS policy exception.   If you have written your own BIRT report provider using the BIRT runtime, then you can set the Allow Origin response header.  If you are using the BIRT viewer webapp then you could avoid this by placing your webapp in the BIRT viewer webapp.  There might be a way to tell the BIRT viewer to set the Allow Origin response header but I don't know what it is.
